qualtran/bloqs/bookkeeping/allocate.py

```diff
@@ -91,10 +93,12 @@ def as_cirq_op(
         self, qubit_manager: 'cirq.QubitManager'
     ) -> Tuple[Union['cirq.Operation', None], Dict[str, 'CirqQuregT']]:
         shape = (*self.signature[0].shape, self.signature[0].bitsize)
-        return (
-            None,
-            {'reg': np.array(qubit_manager.qalloc(self.signature.n_qubits())).reshape(shape)},
+        qubits = (
+            qubit_manager.qborrow(self.signature.n_qubits())
+            if self.dirty
+            else qubit_manager.qalloc(self.signature.n_qubits())
         )
+        return (None, {'reg': np.array(qubits).reshape(shape)})
```

Comment:
Is there any sensible way to unit test these things?

Reply:
That the qubit allocation is happening by calling qalloc / qborrow on the qubit manager when converting a qualtran bloq to a cirq circuit? The updated test in `qualtran/cirq_interop/_bloq_to_cirq_test.py` asserts this.